### PR TITLE
Mark Python 3.12 support

### PR DESCRIPTION
**Pull request by tonybaloney**
I've run the tests locally and all pass

### CODE REVIEW

**Comment by onlined** [comment]
@jiasli @yonzhan @AllyW sorry for mentioning directly, but can someone review this? It's a simple and small PR.

**Comment by jiasli** [comment]
We will work on Python 3.12 support together with https://github.com/Azure/azure-cli/issues/27673.

**Comment by onlined** [comment]
I think this can be merged independently, I don't understand, are these projects coupled? Or don't you trust the tests enough that you can't merge before you run Azure CLI tests with Knack on Python 3.12?

**Comment by jiasli** [comment]
There is much more to do than declaring 3.12 support. See #275 as an example.

File: azure-pipeline.yml [hunk 1]
```diff
@@ -32,6 +32,9 @@ jobs:
         Python311:
           python.version: '3.11'
           tox_env: 'py311'
+        Python312:
+          python.version: '3.12'
+          tox_env: 'py312'
```
Comment (jiasli) [inline azure-pipeline.yml:37]: Could you also update L60 to use `3.12`?

**Review by evelyn-ys — APPROVED** [review]

**Review by bebound — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

File: azure-pipeline.yml (hunk 1 above)
Comment (jiasli) [reply azure-pipeline.yml:37]: I will make the change in #285.

**Pull request merged**